site-src/geps/gep-709.md

```diff
@@ -0,0 +1,161 @@
+These are closely related concepts that can be solved with a new ReferencePolicy
+resource that enables app admins to describe where they trust references from.
+
+## Motivation/User Journeys/Background
+
+This GEP enables the following capabilities:
+
+1. Retaining full control of Gateway and Routes in an infra namespace, while
+   targeting apps in different namespaces.
+1. Traffic splitting between Services in different namespaces.
+1. Mesh overrides to target Services in different namespaces. 
+1. Delegating paths or hostnames to Routes in other namespaces.
+
+## API
+
+Anytime we allow crossing a namespace boundary, we need to be very cautious. To
+ensure that this feature is safe, we need to enforce a handshake mechanism that
+requires resources in both namespaces to agree to this reference. To accomplish
+that, a new ReferencePolicy resource should be introduced.
+
+![Reference Policy](images/709-referencepolicy.png)
+
+With this model, Routes would be able to directly reference Routes and Services
+in other namespaces. These references would only be considered valid if a
+ReferencePolicy in the target namespace explicitly allowed it. 
+
+The following example shows how a HTTPRoute in namespace foo could reference
+a Service in namespace bar. In this example a ReferencePolicy in the bar
+namespace explicitly allows references to Services from HTTPRoutes in the foo
+namespace.
+
+```yaml
+kind: HTTPRoute
+metadata:
+  name: foo
+  namespace: foo
+spec:
+  rules:
+  - matches:
+    - path: /bar
+    forwardTo:
+      backend:
+      - name: bar
+        namespace: bar
+---
+kind: ReferencePolicy
+metadata:
+  name: bar
+  namespace: bar
+spec:
+  from:
+  - group: networking.gateway.k8s.io
+    kind: HTTPRoute
+    namespace: foo
+  to:
+  - group: core
+    kind: Service
+```
+
+### Benefits
+
+* Conceptually similar to NetworkPolicy.
+* A separate resource enables admins to restrict who can allow cross namespace
+  references.
+* Provides consistent way to control references to any resource from a Route.
+* Can be extended in the future for additional use cases.
+* A single ReferencePolicy resource can be used for a namespace in place of
+  separate handshake config on each Service or Route resource.
+
+### Implementation Details
+* `ReferencePolicy.From` and `ReferencePolicy.To` would require at least 1 entry.
+* `Group`, `Kind`, and `Namespace` would all be required fields in
+  `ReferencePolicy.From`.
+* `Group` and `Kind` would both be required fields in `ReferencePolicy.To`.
+* There would be no way to further refine the `From` fields. By trusting a kind
+  of resource in a namespace, you're effectively trusting anyone in that
+  namespace that can administer that kind of resource. Specifying a name does
+  not provide any real protection.
+* Although we may want to consider adding a `Name` field to `ReferencePolicy.To`
+  in the future, it is not clear that it is necessary at this point.
```

Comment:
Can we enumerate which resources are supported in `from.GVK` and those in `to.GVK` in this proposal?

Reply:
I think we need to be clear that this is only for Gateway API implementations, but due to the possibility of custom Route types or any other potential extension points I'm not sure we can have a complete list of allowed resources here.

Reply:
Added guidance for which resources are considered "core" here.